python/lsst/daf/butler/core/location.py

```diff
@@ -141,6 +141,26 @@ def path(self):
         """The path component of the URI."""
         return self._uri.path
 
+    @property
+    def ospath(self):
+        """Path component of the URI localized to current OS."""
+        if self.scheme == 's3':
```

Comment:
We should probably only have ospath return something if we have file or scheme-less. If scheme-less it should remove the path directly and if `file` it should return posix2os.